SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  EditUserDetailsScreen.swift
+//  SwiftAppTemplate
+//
+//  Created by Vijay Goswami on 16/04/24.
+//
+
+import SwiftUI
+
+struct EditUserDetailsScreen: View {
+    @State private var name: String = ""
+    @State private var email: String = ""
+    @State private var selectedGender: Gender = .male
+    @State private var dateOfBirth = Date()
+    @State private var showingDatePicker = false
+    @State private var selectedCountry: String = ""
+    private var selectedLanguage: String {
+           guard let languageCode = Locale.current.languageCode,
+                 let languageName = Locale.current.localizedString(forLanguageCode: languageCode) else {
+               return "English"
+           }
+           return languageName
+       }
+    private var startDate = Calendar.current.date(byAdding: .year, value: -100, to: Date()) ?? Date()
+    @Environment(\.dismiss) var dismiss
+    @State private var showConfirmation: Bool = false
+    @State private var isConfirmationGiven: Bool  = false
+    
+    var body: some View {
+        ZStack {
+            VStack {
+                Header(text: "UpdateUserDetails", hasBackButton: true, onBackArrowClick: {
+                    dismiss()
+                })
+                
+                nameField
+                    .padding(.vertical, 20)
+                
+                emailField
+                
+                genderField
+                    .padding(.vertical, 20)
+                
+                dobField
+                
+                countryField
+                    .padding(.vertical, 20)
+                
+                languageField
+                
+                Spacer()
+                
+                TextButton(onClick: {
+                    if hasEnteredAllDetails() {
+                        showConfirmation = true
+                    }
+                }, text: "Update")
+            }
+            .padding(20)
+           
+            Spacer()
+            
+            bottomSheet
+        }
+        .onChange(of: isConfirmationGiven) { isConfirmationGiven in
+            if(isConfirmationGiven){
+                saveDetails()
+                dismiss()
+            }
+        }
+        .onAppear(){
+            initializeDetails()
+        }
+     
+    }
+    
+    private var nameField : some View {
```

Comment:
All these fields like nameField and emailField could be one struct, genderField and countryField could be one struct.

Reply:
fixed, gender and country fields have different Stack view Vstack and Hstack